Patch-Source: https://github.com/mypaint/mypaint/commit/fb6f7d6b12238d125294496b105f5f36e658bc61
---
From fb6f7d6b12238d125294496b105f5f36e658bc61 Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Sun, 13 Sep 2026 22:55:19 +0300
Subject: [PATCH] Fix SWIG 4.5 compatibility

Replace old PyString_FromString call with PyUnicode_FromString. SWIG 4.5 removed its Python 2 compatibility aliases, exposing the undefined C API name while generating bindings for Python 3.
---
 lib/tiledsurface.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/tiledsurface.hpp b/lib/tiledsurface.hpp
index 315f0d9b7..16b5bcc8b 100644
--- a/lib/tiledsurface.hpp
+++ b/lib/tiledsurface.hpp
@@ -133,7 +133,7 @@ class TiledSurface : public Surface {
 static PyObject *
 get_module(char *name)
 {
-    PyObject *pName = PyString_FromString(name);
+    PyObject *pName = PyUnicode_FromString(name);
     PyObject *pModule = PyImport_Import(pName);
     Py_DECREF(pName);
 
